src/main/java/edu/ucsb/cs156/gauchoride/controllers/DriverAvailabilityController.java

```diff
@@ -64,5 +64,40 @@ public DriverAvailability postDriverAvailability(
         DriverAvailability savedDriverAvailability = driverAvailabilityRepository.save(driverAvailability);
 
         return savedDriverAvailability;
+        }
+
+    @Operation(summary= "Delete Driver Availability")
+    @PreAuthorize("hasRole('ROLE_DRIVER')")
+    @DeleteMapping("")
+    public Object deletemenuitem(
```

Comment:
Drivers should only be able to delete their own availabilities, and admins should be able to delete anyones

Reply:
Drivers are still able to whatever availability, you need to check which role the user has inside the method and based on that determine whether they can delete or not

Please see existing controller https://github.com/ucsb-cs156/proj-gauchoride/blob/main/src/main/java/edu/ucsb/cs156/gauchoride/controllers/RideController.java